dotnetv3/Keyspaces/Tests/KeyspacesWrapperTests.cs

```diff
@@ -0,0 +1,171 @@
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("testsettings.json") // Load test settings from .json file.
+                .AddJsonFile("testsettings.local.json",
+                    true) // Optionally load local settings.
+                .Build();
+
+            _client = new AmazonKeyspacesClient();
+            _wrapper = new KeyspacesWrapper(_client);
+
+            _keyspaceName = _configuration["KeyspaceName"];
+            _tableName = _configuration["TableName"];
+
+            // Define the schema for the test table.
+            var allColumns = new List<ColumnDefinition>
+            {
+                new ColumnDefinition { Name = "title", Type = "text" },
+                new ColumnDefinition { Name = "year", Type = "int" },
+                new ColumnDefinition { Name = "release_date", Type = "timestamp" },
+                new ColumnDefinition { Name = "plot", Type = "text" },
+            };
+
+            var partitionKeys = new List<PartitionKey>
+            {
+                new PartitionKey { Name = "year", },
+                new PartitionKey { Name = "title" },
+            };
+
+            _schemaDefinition = new SchemaDefinition
+            {
+                AllColumns = allColumns,
+                PartitionKeys = partitionKeys,
+            };
+        }
+
+        [Fact()]
+        [Order(1)]
+        [Trait("Category", "Integration")]
+        public async Task CreateKeyspaceTest()
+        {
+            _keyspaceArn = await _wrapper.CreateKeyspace(_keyspaceName);
+
+            // Wait until keyspace has been created.
+            Thread.Sleep(10000);
+            var getKeyspaceArn = "";
+            do
+            {
+                getKeyspaceArn = await _wrapper.GetKeyspace(_keyspaceName);
+            } while (getKeyspaceArn != _keyspaceArn);
+            Assert.NotNull(_keyspaceArn);
+        }
+
+        [Fact()]
+        [Order(2)]
+        [Trait("Category", "Integration")]
+        public async Task GetKeyspaceTest()
+        {
+            var keyspaceArn = await _wrapper.GetKeyspace(_keyspaceName);
+            Assert.Equal(keyspaceArn, _keyspaceArn);
+        }
+
+        [Fact()]
+        [Order(3)]
+        [Trait("Category", "Integration")]
+        public async Task CreateTableTest()
+        {
+            var tableArn = await _wrapper.CreateTable(_keyspaceName, _schemaDefinition, _tableName);
+
+            // Wait for the table to be created.
+            Thread.Sleep(10000);
+            try
+            {
+                GetTableResponse resp;
+
+                // Wait for the table to be created.
+                do
+                {
+                    resp = await _wrapper.GetTable(_keyspaceName, _tableName);
+                } while (resp.Status != TableStatus.ACTIVE);
+            }
+            catch (ResourceNotFoundException ex)
+            {
+                Console.WriteLine($"Error: {ex.Message}");
```

Comment:
Do you need an assert within this catch? Or will tableArn == null if we get this error?

Reply:
Actually it is necessary. You have to give the table time to be created and before that, it will raise an error that the table doesn't exist. The same can happen for CreateKeyspace.